kubeflow/metacontroller/metacontroller.libsonnet

```diff
@@ -0,0 +1,186 @@
+{
+  local k8s = import "k8s.libsonnet",
+  local util = import "kubeflow/core/util.libsonnet",
+  local crd = k8s.apiextensions.v1beta1.customResourceDefinition,
+
+  new(_env, _params):: {
+    local params = _env + _params {
```

Comment:
nit: should we just get namespace from env and not params?

The only reason we ever got it from params was because the early version of ksonnet didn't support getting it from env.

I think this snippet just keeps getting copied around. I think we should try to stick with the pattern env maps to namespace.

I think if we start letting different components deploy in different namespaces its confusing to users so I think we should treat that as an exception.